server/src/main/java/org/opensearch/gateway/GatewayMetaState.java

```diff
@@ -767,7 +782,7 @@ private boolean shouldWriteFullClusterState(ClusterState clusterState, int codec
             assert lastAcceptedManifest == null || lastAcceptedManifest.getCodecVersion() <= codecVersion;
             if (lastAcceptedState == null
                 || lastAcceptedManifest == null
-                || lastAcceptedState.term() != clusterState.term()
+                || (remoteClusterStateService.isRemotePublicationEnabled() == false && lastAcceptedState.term() != clusterState.term())
```

Comment:
what is the purpose of publication check here? if publication is disabled why would this method be called?

Reply:
It is still possible that we have remote state is enabled with publication feature disabled. In case remote state is enabled, it will still be possible the lastAcceptedState is non null and stale, which is status quo as previously.